autonick.lua

```diff
@@ -0,0 +1,56 @@
+-- automatically nickname dwarves
+--[====[
+
+autonick
+========
+Automatically nickname dwarves according to ``dfhack-config/autonick.txt``.
+
+One nickname per line.
+Empty lines, lines beginning with ``#`` and repeat entries are discarded.
+
+Dwarves with manually set nicknames are ignored.
+
+If there are fewer available nicknames than dwarves, the remaining
+dwarves will go un-nicknamed.
+
+You may wish to use this script with the "repeat" command, e.g:
+``repeat -name autonick -time 3 -timeUnits months -command [ autonick ]``
+
+]====]
+
+names = {};
+
+path = dfhack.getDFPath () .. "/dfhack-config/autonick.txt";
+
+-- grab list, put in array
+for line in io.lines(path) do
+    if (line:trim() ~= "") and (line:sub(1, 1) ~= "#") then
```

Comment:
it would be better to assign the trimmed value back to line -- otherwise the space-padded version will get inserted into the names list.

also, line:startswith('#') also works and is shorter